### PR TITLE
[e2e] Add LB tests about switching between public and internal ones

### PR DESCRIPTION
**Pull request by lzhecheng**
Signed-off-by: Zhecheng Li <zhechengli@microsoft.com>

#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Add 2 LB tests:
* internal -> public
* public -> internal with specific IP
This change is to replace an upstream k/k test because there's a bug
there now.
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: tests/e2e/network/ensureloadbalancer.go [hunk 1]
```diff
@@ -305,6 +305,97 @@ var _ = Describe("Ensure LoadBalancer", func() {
...
+		Expect(err).NotTo(HaveOccurred())
+		utils.Logf("Successfully created LoadBalancer service %s in namespace %s", testServiceName, ns.Name)
+		defer func() {
+			By("Cleaning up")
+			err = utils.DeleteService(cs, ns.Name, testServiceName)
+			Expect(err).NotTo(HaveOccurred())
+		}()
+
+		By("Waiting for exposure of the internal Service")
+		_, err = utils.WaitServiceExposureAndValidateConnectivity(cs, ns.Name, testServiceName, "")
```
Comment (MartinForReal) [inline tests/e2e/network/ensureloadbalancer.go:325]: I'm wondering if we could check if ip is updated and new connection is sucessful.

Comment (lzhecheng) [reply tests/e2e/network/ensureloadbalancer.go:350]: Updated. PTAL.

File: tests/e2e/utils/pod_utils.go [hunk 2]
```diff
@@ -136,6 +136,40 @@ func CreatePod(cs clientset.Interface, ns string, manifest *v1.Pod) error {
...
+	pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: ns,
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name:            "agnhost",
+					Image:           "e2eteam/agnhost:2.26",
```
Comment (feiskyer) [inline tests/e2e/utils/pod_utils.go:152]: Do we have this image in MCR or GCR?

Comment (lzhecheng) [reply tests/e2e/utils/pod_utils.go]: Updated. Using gcr now.

**Pull request merged**